src/umpire/ResourceManager.cpp

```diff
@@ -250,6 +267,26 @@ ResourceManager::registerAllocator(const std::string& name, Allocator allocator)
   m_allocators_by_name[name] = allocator.getAllocationStrategy();
 }
 
+void
+ResourceManager::removeAllocator(Allocator allocator)
+{
```

Comment:
I think we need to discuss how this would work. I'm not sure ripping out all the Allocators sharing a strategy is a good idea.

Reply:
You're right, I should compare by ID instead of by strategy.

Reply:
Wait, `m_allocators_by_name` only has strategies not allocators, same with `m_allocators_by_id`. If they instead held allocators then I could could compare the allocators by ID.

Reply:
Allocator ID == strategy ID. What I mean is, we need two different methods: one that "deregisters" a named AllocationStrategy/Allocator, the second that destroys it if and only if it's not registered under another name.

Reply:
That makes sense to me. This isn't actually integral to this PR, I didn't wind up using it anywhere seeing as the Allocators/AllocationStrategies are now all "destroyed" upon calling `ResourceManager::finalize()`. However I certainly think this functionality is useful for destroying AllocationStrategies that themselves hold a significant amount of data.

Reply:
Done.